plugins/modules/cluster_role.py

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2019, Paul Arthur <paul.arthur@flowerysong.com>
+# Copyright: (c) 2019, XLAB Steampunk <steampunk@xlab.si>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'XLAB Steampunk'}
+
+DOCUMENTATION = '''
+module: cluster_role
+author:
+ - Paul Arthur (@flowerysong)
+ - Manca Bizjak (@mancabizjak)
+ - Aljaz Kosir (@aljazkosir)
+ - Tadej Borovsak (@tadeboro)
+short_description: Manages Sensu cluster roles
+description:
+  - For more information, refer to the Sensu documentation at
+    U(https://docs.sensu.io/sensu-go/latest/reference/rbac/)
+extends_documentation_fragment:
+  - sensu.sensu_go.auth
+  - sensu.sensu_go.name
+  - sensu.sensu_go.state
+notes:
+  - Parameter C(auth.namespace) is ignored in this module.
+options:
+  rules:
+    description:
+      - Rules that the cluster role applies.
+    type: list
+    suboptions:
+      verbs:
```

Comment:
Same question as with rules: are empty lists OK?

Reply:
Sensu API seems to be OK with empty list, and Ansible too..., but to me it just doesn't make sense. This is a valid rule:
```
{'verbs': [], 'resources': [], 'resource_name': []}
```
Object saved on the API looks lie this:
```
{
    "metadata": {
        "name": "name"
    },
    "rules": [
        {
            "resource_names": null,
            "resources": null,
            "verbs": null
        }
    ]
}
```
Should we check for this kind of stuff in a custom validator too? Atleast for `verbs` and `resources` since they required on the API.
This applies to all three keys.

Reply:
I also noticed this behavior when implementing the `role` module. At that time, I didn't think about implementing custom validation to check whether `verbs` and `resources` are empty lists precisely because of what you reported, i.e. because the API seems to allow this case.

Nonetheless I am in favor of adding additional validations here and in `role` despite the API not breaking because of such cases (even when the docs state otherwise).

Reply:
I asked upstream devs about this. I would say that for now, we keep things as they are (`[]` is a valid value for verbs, resources and resource names) and change them once we hear back from the sensu gals and guys. Does this sound OK with you, @mancabizjak and @aljazkosir?